gdbgui-mpi/compile.sh

```diff
@@ -0,0 +1,3 @@
+#! /bin/bash
```

Comment:
I think the `gdbgui-mpi` folder might fit better in `examples/mpi`. There are already examples in the `examples` folder.